Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,217 @@
+         * @method _makeBlocksActive
+         * @protected
+         */
+        _makeBlocksActive: function () {
+            var blocks = this.get('blocks');
+
+            Object.keys(blocks).forEach(function (key) {
+                blocks[key].set('active', true);
+            });
+        },
+
+        /**
+         * Renders blocks on the dashboard
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this._sortBlocks(this.get('blocks')).forEach(Y.bind(function (block) {
+                var targets = block.getTargets();
+
+                if (targets.indexOf(this) === -1) {
+                    block.addTarget(this);
+                }
+
+                fragment.append(block.render().get('container'));
+            }, this));
+
+            this.get('container').one(SELECTOR_CONTENT).setHTML(fragment);
+        },
+
+        /**
+         * Sorts the provided hash of blocks either by priority
+         * or alphabetically when both blocks have the same priority.
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Object} hash of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            var list = [];
+
+            Object.keys(blocks).forEach(function (key) {
+                list.push(blocks[key]);
+            });
+
+            list.sort(function (a, b) {
+                var priorityA = a.get('priority'),
+                    priorityB = b.get('priority');
+
+                if (priorityA === priorityB) {
+                    // sort alphabetically (from A to B)
+                    return a.get('identifier') < b.get('identifier');
+                } else {
+                    // sort by priority (max to 0)
+                    return priorityA > priorityB;
+                }
+            });
+
+            return list;
+        },
+
+        /**
+         * Verifies provided `blocks` attribute value
+         * by checking whether provided values are dashboard block view instances
+         *
+         * @method _checkBlocks
+         * @protected
+         * @param value {Any} new vaue of `blocks` attribute
+         * @return {Object}
+         */
+        _checkBlocks: function (value) {
+            var blocks = {};
+
+            if (typeof value !== 'object' || Array.isArray(value)) {
+                return blocks;
+            }
+
+            Object.keys(value).forEach(function (key) {
+                if (value[key] instanceof Y.eZ.BaseDashboardBlockView) {
+                    blocks[key] = value[key];
+                }
+            });
+
+            return blocks;
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param view {Y.eZ.BaseDashboardBlockView} dashboard block view instance
```

Comment:
In this case this might lead to errors. See the block-based dashboard view is named `eZ.DashboardBlockBasedView` and the base dashboard block view you want to name: `eZ.DashboardBlockBaseView`